nengo/learning_rules.py

```diff
@@ -149,54 +187,80 @@ class BCM(LearningRuleType):
+    post_synapse : Synapse
+        Synapse model used to filter the post-synaptic activities
+        (see `~nengo.synapses.Synapse`).
+    pre_synapse : Synapse
+        Synapse model used to filter the pre-synaptic activities
+        (see `~nengo.synapses.Synapse`).
+    theta_synapse : Synapse
+        Synapse model used to filter the theta signal
+        (see `~nengo.synapses.Synapse`).
     """
 
     modifies = 'weights'
     probeable = ('theta', 'pre_filtered', 'post_filtered', 'delta')
 
-    pre_tau = NumberParam('pre_tau', low=0, low_open=True)
-    post_tau = NumberParam('post_tau', low=0, low_open=True)
-    theta_tau = NumberParam('theta_tau', low=0, low_open=True)
-
-    def __init__(self, pre_tau=0.005, post_tau=None, theta_tau=1.0,
-                 learning_rate=1e-9):
-        self.theta_tau = theta_tau
-        self.pre_tau = pre_tau
-        self.post_tau = post_tau if post_tau is not None else pre_tau
+    learning_rate = NumberParam(
+        'learning_rate', low=0, readonly=True, default=1e-9)
+    pre_synapse = SynapseParam(
+        'pre_synapse', default=Lowpass(tau=0.005), readonly=True)
+    post_synapse = SynapseParam(
+        'post_synapse', default=None, readonly=True)
+    theta_synapse = SynapseParam(
+        'theta_synapse', default=Lowpass(tau=1.0), readonly=True)
+
+    pre_tau = _deprecated_tau("pre_tau", "pre_synapse")
+    post_tau = _deprecated_tau("post_tau", "post_synapse")
+    theta_tau = _deprecated_tau("theta_tau", "theta_synapse")
+
+    def __init__(self, learning_rate=Default, pre_synapse=Default,
+                 post_synapse=Default, theta_synapse=Default,
+                 pre_tau=Unconfigurable, post_tau=Unconfigurable,
+                 theta_tau=Unconfigurable):
         super(BCM, self).__init__(learning_rate, size_in=0)
 
+        if pre_tau is Unconfigurable:
+            self.pre_synapse = pre_synapse
+        else:
+            self.pre_tau = pre_tau
+
+        if post_tau is Unconfigurable:
+            self.post_synapse = (self.pre_synapse if post_synapse is Default
+                                 else post_synapse)
+        else:
+            self.post_tau = post_tau
+
+        if theta_tau is Unconfigurable:
+            self.theta_synapse = theta_synapse
+        else:
+            self.theta_tau = theta_tau
+
     @property
     def _argreprs(self):
-        args = []
-        if self.pre_tau != 0.005:
-            args.append("pre_tau=%g" % self.pre_tau)
-        if self.post_tau != self.pre_tau:
-            args.append("post_tau=%g" % self.post_tau)
-        if self.theta_tau != 1.0:
-            args.append("theta_tau=%g" % self.theta_tau)
-        if self.learning_rate != 1e-9:
-            args.append("learning_rate=%g" % self.learning_rate)
-        return args
+        return (('learning_rate', BCM.learning_rate.default),
+                ('pre_synapse', BCM.pre_synapse.default),
+                ('post_synapse', self.pre_synapse),
```

Comment:
I think this is a typo and should be ` ('post_synapse', BCM.post_synapse.default)` at least for the sake of consistency?

Reply:
The `post_synapse` defaults to the value of `pre_synapse`, so that's the value we want to check against (to see if the user set it to something other than the default).